src/pages/blog/dynamic-components-in-your-mpa.mdx

```diff
@@ -0,0 +1,248 @@
+
+Lit & Stencil are both built with the full structure & safety of the web-component architecture in mind, but luckily they both have escape hatches to avoid things like the shadow-DOM. With Lit, you just need to override the `createRenderRoot` method in your component. Normally this method returns a shadow-root but instead you can have it return `this` because custom-element classes are, themselves, DOM elements that extend the `HTMLElement` class. In Stencil, you only need to add the `shadow: false` option to the configuration object passed to your `@Component` decorator. This means that with very little work both of these tools can be made to support the architecture of most projects as needed, but they are not without unique considerations.
+
+## Scenario 4: You've been prescribed a solution.
+
+We've all been there: Your project manager tells you that another development team has already built the component in React. They even bundled it into a library, so all you need to do is drop in the library right? But you know it's not that easy. You're not using React. You're not even really using a frontend framework. So how can we make this process of utilizing React in your MPA as painless as possible?
+
+### Bundling a Vite app into an MPA
+
+There are steps that just can't be skipped surrounding the bundling, building, and exporting process. In my experience, I've found that utilizing Vite is the best way to go these days. It solves the most problems up front, has the fastest performance, and requires very little in terms of configuration overrides.
+
+If you instantiate a Vite application inside your MPA repository, you will naturally get a `vite.config.js` file. In the Vite config, if you set `config.build.rollupOptions.output.manualChunks` to `undefined` and set `config.build.rollupOptions.output.entryFileNames` to something simple like `"app.js"`, you will remove all chunking & file-name hashing from the build output so that your JS bundle will always be one file with a consistent name. This makes it much easier to link to from your MPA.
```

Comment:
Maybe add a note about making sure your MPA handles cacheing of any potential changes to `app.js` accordingly.

Reply:
Done!